src/client-c++/cli_main.cpp

```diff
@@ -305,84 +306,89 @@ int main(int argc, char** argv)
+        sSfServerV1.mPasswordPtr    = NULL;
 
         sSfArgsV1.mProject          = sArgs.mProject;
         sSfArgsV1.mComment          = sArgs.mComment;
         sSfArgsV1.mExtraServerParms = sArgs.mExtraServerParms;
     }
 
-    if(sIsSuccess && !sArgs.mPasswordEnvVar.empty())
+    if(sIsSuccess &&
+       (NULL != getenv("SSH_AUTH_SOCK") ||
+        NULL != getenv("SSH_AGENT_PID")) )
+    {
+        sf_client::rc sRc        = sf_client::failure;
+        // Attempt a connection without providing a password. If the key is registered in an agent
+        // then the connection will work without needing user interaction.
+        sRc = sf_client::connectToServer(sSfServerV1, sSession);
+
+        sIsConnected = (sf_client::success == sRc);
+    }
+
+    // Provide private key to any other connection attempts
+    sSfServerV1.mPrivateKeyPath = sArgs.mPrivateKeyPath;
+
+    if(sIsSuccess && !sIsConnected && !sArgs.mPasswordEnvVar.empty())
     {
-        std::cout << "Running with ENV" << std::endl;
         sSfServerV1.mPasswordPtr = getenv(sArgs.mPasswordEnvVar.c_str());
         sf_client::rc sRc        = sf_client::connectToServer(sSfServerV1, sSession);
         sIsSuccess               = sf_client::success == sRc;
         // The password was provided in the environment. If it is wrong there is nothing else to be
         // done.
     }
-    else if(sIsSuccess)
+    else if(sIsSuccess && !sIsConnected)
     {
-        std::cout << "Attempt running with agent" << std::endl;
-        // Attempt a connection without providing a password. If the key is registered in an agent
-        // then the connection will work without needing user interaction.
-        sSfServerV1.mPasswordPtr = NULL;
-        sf_client::rc sRc        = sf_client::connectToServer(sSfServerV1, sSession);
-        std::cout << sRc << std::endl;
 
-        if(sf_client::success != sRc)
-        {
-            std::cout << "Running with prompt" << std::endl;
-            sRc = sf_client::success;
+        sf_client::rc sRc = sf_client::success;
 
-            char* sPasswordPtr = (char*)OPENSSL_malloc(MaxPasswordSize);
-            if(sPasswordPtr)
-                memset(sPasswordPtr, 0, MaxPasswordSize);
+        char* sPasswordPtr = (char*)OPENSSL_malloc(MaxPasswordSize);
+        if(sPasswordPtr)
+            memset(sPasswordPtr, 0, MaxPasswordSize);
 
-            if(!sPasswordPtr)
-            {
-                std::cerr << "Unable to allocate memory for password buffer" << std::endl;
-                sIsSuccess = false;
-            }
-            else
+        if(!sPasswordPtr)
+        {
+            std::cerr << "Unable to allocate memory for password buffer" << std::endl;
+            sIsSuccess = false;
+        }
+        else
+        {
+            std::size_t sNumAttempts = 0;
+            do
             {
-                std::size_t sNumAttempts = 0;
-                do
+                sNumAttempts++;
+
+                std::size_t sPasswordLength = 0;
```

Comment:
needs to be uint64_t for compatibility across different definitions of std::size_t (GetPassword takes uint64_t)